common/djangoapps/util/help_desk.py

```diff
@@ -0,0 +1,443 @@
+"""
+Module containing helps desk factory function and help desk interfaces.
+"""
+import json
+import logging
+from abc import abstractmethod
+from importlib import import_module
+
+import requests
+import zendesk
+from django.conf import settings
+from django.core.cache import caches
+from opaque_keys.edx.keys import CourseKey
+
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from student.models import CourseEnrollment
+
+log = logging.getLogger(__name__)
+
+
+class HelpDeskService(object):
+    """
+    Class containing the common logical for the help desk functionallity.
+    """
+
+    @abstractmethod
+    def get_help_desk_service(self):
+        """
+        Returns the configured help desk service depending on HELP_DESK_SERVICE_BACKEND value.
+        """
+        help_desk_service = getattr(settings, 'HELP_DESK_SERVICE_BACKEND', '')
+
+        if not help_desk_service:
+            raise ValueError('HELP_DESK_SERVICE_BACKEND')
+
+        module_string = help_desk_service.split(':')
+        help_desk_class_string = module_string[-1]
+        help_desk_module = import_module(module_string[0])
+
+        return getattr(help_desk_module, help_desk_class_string)
+
+
+class FreshdeskError(Exception):
+    """
+    FreshdeskError base exception.
+    """
+    def __init__(self, msg, code, response):
+        self.msg = msg
+        self.error_code = code
+        self.response = response
+
+    def __str__(self):
+        return repr('%s: %s %s' % (self.error_code, self.msg, self.response))
+
+
+class _FreshdeskApi(object):
+    """
+    Freshdesk API implementation.
+    """
+
+    CACHE_PREFIX = 'FRESHDESK_API_CACHE'
+    CACHE_TIMEOUT = 60 * 60
+
+    def __init__(self):
+        """
+        Instantiate the Freshdesk API.
+
+        `HELPDESK_URL` and `HELPDESK_API_KEY` must be set
+        Set `HELPDESK` = 'Freshdesk'
+        in `django.conf.settings`.
+        """
+
+
+    def create_ticket(self, subject, desc, name, email, priority=1, status=2, group_id=None):
+        """
+        Create the given `ticket` in Freshdesk.
+
+        The ticket should have the format specified by the freshdesk api.
+        https://freshdesk.com/api#create_ticket
+        """
+        headers = {'Content-Type': 'application/json'}
+        payload = {
+            'subject': subject,
+            'description': desc,
+            'name': name,
+            'email': email,
+            'priority': priority,
+            'status': status,
+            'group_id': group_id
+        }
+
+        response = requests.post(
+            settings.HELPDESK_URL + '/api/v2/tickets',
+            auth=(settings.HELPDESK_API_KEY, "unused_but_required"),
+            headers=headers,
+            data=json.dumps(payload),
+            allow_redirects=False)
+
+        if 200 <= response.status_code <= 299:
```

Comment:
It there is no particular reason to check that interval of statuses, let's use 
```suggestion
        if response.ok:
```

See reference here https://2.python-requests.org//en/master/api/#requests.Response.ok

Reply:
I don't know, it's the edX implementation, addressed.